tests/test_forms.py

```diff
@@ -1493,3 +1494,121 @@ async def test_submit(form_class: Type[FormAction]):
 def test_form_deprecation():
     with pytest.warns(FutureWarning):
         FormAction()
+
+
+class TestFormValidationAction(FormValidationAction):
+    def name(self) -> Text:
+        return "some_form"
+
+    def validate_slot1(
+        self,
+        slot_value: Any,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> Dict[Text, Any]:
+        if slot_value == "correct_value":
+            return {
+                "slot1": "validated_value",
+            }
+
+    def validate_slot2(
+        self,
+        slot_value: Any,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> Dict[Text, Any]:
+        if slot_value == "correct_value":
+            return {
+                "slot2": "validated_value",
+            }
+
+    async def validate_slot3(
+        self,
+        slot_value: Any,
+        dispatcher: "CollectingDispatcher",
+        tracker: "Tracker",
+        domain: "DomainDict",
+    ) -> Dict[Text, Any]:
+        if slot_value == "correct_value":
+            return {
+                "slot3": "validated_value",
+            }
+
+
+async def test_form_validation_action():
+    form = TestFormValidationAction()
+
+    # tracker with active form
+    tracker = Tracker(
+        "default",
+        {},
+        {},
+        [SlotSet("slot1", "correct_value"), SlotSet("slot2", "incorrect_value")],
+        False,
+        None,
+        {"name": "some_form", "validate": True, "rejected": False},
+        "action_listen",
+    )
+
+    dispatcher = CollectingDispatcher()
+    events = await form.run(dispatcher=dispatcher, tracker=tracker, domain=None)
+    # check that the form was activated and validation was performed
```

Comment:
this comment doesn't make sense as the form was already activated (same for test functions)
```suggestion
```